device/transport/http/src/http.ts

```diff
@@ -153,7 +153,7 @@ export class Http extends EventEmitter implements Client.Transport {
           /*Codes_SRS_NODE_DEVICE_HTTP_16_033: [if the `getDeviceCredentials` fails with an error, the Http request shall call its callback with that error]*/
           done(err);
         } else {
-          const path = endpoint.eventPath(config.deviceId);
+          const path = endpoint.eventPath(encodeUriComponentStrict(config.deviceId));
```

Comment:
>eventPath(encodeUriComponentStrict(config.deviceId)); [](start = 32, length = 53)

I had a similar change in the modules-preview branch.  I feel vindicated.

Reply:
Should we be using this strict version everywhere?  I didn't even know we had this until now and was using encodeUriComponent because I see it everywhere 

---
In reply to: [189098076](https://github.com/Azure/azure-iot-sdk-node/pull/292#discussion_r189098076) [](ancestors = 189098076)